rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1435,6 +1435,16 @@ public void addSignature(BtcECKey federatorPublicKey, List<byte[]> signatures, b
         }
 
         Federation federation = optionalFederation.get();
+        Optional<FederationMember> federationMember = federation.getMemberByBtcPublicKey(federatorPublicKey);
+        if (!federationMember.isPresent()){
```

Comment:
I was discussing with @julia-zack  that maybe the correct thing will be to throw an exception in case no federation member is found; this is already covered in the above `if.` If a federation that has a federation member that matches the given btc public key is found. It means it will always find a federation member by the given btc public key.

So, the proposal change will be to throw a `IllegalStateException` to satisfy Sonar and be protected if it happens for some reason.

Like this:

```
FederationMember signingFederationMember = federation.getMemberByBtcPublicKey(federatorPublicKey)
            .orElseThrow(() -> new IllegalStateException(
                String.format("Supplied federator public key %s does not belong to any of the federators.", federatorPublicKey)
            ));
```

Reply:
Yes, I think the right thing to do is throw an exception, and it should be specific for this case.
It would be really inconsistent if a federation is found by searching for some btc public key but then there is no member that actually has that btc public key.

Reply:
But if in the future we receive the `Federation` as a parameter, we still will have to check if the btc public key we want to sign is part of it, right?
At the point of getting the federation member by the btc public key we already "know" that we will find it.

Im thinking on a new option... if we want to search for a federation and a federation member using one btc public key, this should an "if and only if" situation (if we find the federation, we find the member and viceversa). So maybe we could do something like this:

```
Optional<Federation> optionalFederation = getFederationFromPublicKey(federatorPublicKey);
Optional<FederationMember> federationMember = federation.getMemberByBtcPublicKey(federatorPublicKey);
if (!optionalFederation.isPresent() || !federationMember.isPresent()) {
           logger.warn(
                "Supplied federator public key {} does not belong to any of the federators.",
                federatorPublicKey
            );
            return;
}
```


wdyt? @nathanieliov @marcos-iov

Reply:
@julia-zack It doesn't make sense to search for `federationMember` if you don't find a `federation`; that's why the "if and only if" situation does not apply in this case since federationMember is dependent on a Federation being found.

I'll go with the current solution and add tests for this new flow(basically after `RSKIP415`.

Reply:
Fixed!

Reply:
It does apply. You don't expect to find the federation if you don't find the fed member either. The only reason that finding the fed member "is dependent" on finding the federation is because the code is written in that order. But if you reverse the order, the dependence will be the other way around.

Actually, the implementation of `getFederationFromPublicKey` searches if any fed member has that public key

Reply:
Chenging the log message is probably the best yes. Also, @nathanieliov remember to add `[FUNCTION_NAME]` as a prefix to the logs

Reply:
I agree with both! Thanks guys for the feedback!

Reply:
Logs updated!